### PR TITLE
wpcom merge: publicize

### PR DESCRIPTION
**Pull request by dereksmart**
This brings the publicize files in sync with wordpress.com. 

Publicize_Util is no longer needed, via r174489-wpcom

The new `publicize_calypso_url()` was added to wpcom in r128505-wpcom, but is not used anywhere in Jetpack.

### CODE REVIEW

**Comment by nb** [comment]
@nebososo, can you have a look, please?

**Comment by dereksmart** [comment]
@nebososo Thank you! I looked too quickly at r174489 and only noticed the namespaced `\Publicize\ExtractorUtils` substitutions.  

The changes made to publicize/publicize.php will be fine to bring over, and we can remove this class 👍

**Comment by dereksmart** [comment]
@nebososo Updated to include the rest of the changes. Mind taking a quick look?

File: modules/publicize/publicize.php [hunk 1]
```diff
@@ -565,4 +565,33 @@ function test_publicize_conns() {
...
+				continue;
+			}
+			$search[] = "%$arg%";
+			$replace[] = "%$k\$s";
+		}
+		return str_replace( $search, $replace, $string );
+	}
+}
+
+function publicize_calypso_url() {
```
Comment (dereksmart) [inline modules/publicize/publicize.php]: This was added in r128505-wpcom, but is not used anywhere in Jetpack.  Bringing over here just for consistency in files.

**Comment by nebososo** [comment]
Looks good. I also cloned it and ran a couple of quick tests and everything seems to be working.

**Pull request merged**